### PR TITLE
[monorepo] fix web_long_running_tests for monorepo

### PR DESCRIPTION
**Pull request by yjbanov**
The realm file is now populated on presubmit, and the old logic would skip the `web_long_running_tests` shard. The new logic allows the tests to always run with one important exception: the CDN test can only run in post-submit because presubmit binaries never go to the CDN.

### CODE REVIEW

File: dev/bots/suite_runners/run_web_tests.dart [hunk 1]
```diff
@@ -91,14 +91,15 @@ class WebTestsSuite {
...
   Future<void> webLongRunningTestsRunner() async {
-
     final String engineVersionFile = path.join(flutterRoot, 'bin', 'internal', 'engine.version');
-    final String engineRealmFile = path.join(flutterRoot, 'bin', 'internal', 'engine.realm');
     final String engineVersion = File(engineVersionFile).readAsStringSync().trim();
+
+    // The realm file is only populated in presubmit because presubmit engine binaries
+    // are not uploaded to the prod GCS buckets.
+    final String engineRealmFile = path.join(flutterRoot, 'bin', 'internal', 'engine.realm');
     final String engineRealm = File(engineRealmFile).readAsStringSync().trim();
```
Comment (jtmcdole) [inline dev/bots/suite_runners/run_web_tests.dart:100]: if we had wrapped this in a try/catch and assumed "" on errors, we would have saved ourselves so much time.

File: dev/bots/suite_runners/run_web_tests.dart [hunk 2]
```diff
@@ -91,14 +91,15 @@ class WebTestsSuite {
...
-    if (engineRealm.isNotEmpty) {
-      return;
-    }
+    final bool isPostSubmit = engineRealm.isEmpty;
```
Comment (jtmcdole) [inline dev/bots/suite_runners/run_web_tests.dart:101]: MQ and PostSubmit have the realm as empty.

File: dev/bots/suite_runners/run_web_tests.dart [hunk 3]
```diff
@@ -206,10 +207,14 @@ class WebTestsSuite {
       () => _runWebDebugTest('lib/stack_trace.dart'),
       () => _runWebDebugTest('lib/framework_stack_trace.dart'),
       () => _runWebDebugTest('lib/web_directory_loading.dart'),
-      () => _runWebDebugTest('lib/web_resources_cdn_test.dart',
-        additionalArguments: <String>[
-          '--dart-define=TEST_FLUTTER_ENGINE_VERSION=$engineVersion',
-        ]),
+
+      // Only run the CDN test in post-submit because presubmit binaries never go to the CDN.
```
Comment (jtmcdole) [inline dev/bots/suite_runners/run_web_tests.dart:211]: same here; post submit && merge queue

**Review by jtmcdole — APPROVED** [review]
lgtm

**Comment by yjbanov** [comment]
Closing for now. These tests are not passing in presubmit, and the fix is probably not as simple as what's in this PR.

**Pull request closed without merging**